### PR TITLE
Validate regex csquery

### PR DESCRIPTION
**Pull request by hemdagem**
addresses issue #138

Made use of ValidationAttributes to validate csquery and regex.

@yetanotherchris  @irmbrady  @Workshop2

### CODE REVIEW

**Comment by Workshop2** [comment]
Does this work if you reference variables in the regex?

**Comment by Workshop2** [comment]
@hemdagem <bump> don't forget about testing for variables :+1:

File: src/Syringe.Web/ValidationAttributes/AssertionTypeValidationAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                return attribute.GetValidationResult(value, context);
+            }
+            if (model.AssertionMethod == AssertionMethod.Regex)
+            {
+                ValidRegexAttribute attribute = new ValidRegexAttribute { ErrorMessage = "Invalid Regex" };
+                return attribute.GetValidationResult(value, context);
+            }
+
+            //no assertion type so always valid
+            return ValidationResult.Success;
```
Comment (Workshop2) [inline src/Syringe.Web/ValidationAttributes/AssertionTypeValidationAttribute.cs]: Shouldn't we fail it by default? Or are there valid assertion types where we would want to `success` it?

**Comment by hemdagem** [comment]
Yep need to update this. I guess when the value gets checked I need to convert the variables to the actual value and then test that field to see of the regex/csquery passes. Also display correct messaging if a variable is invalid.

**Pull request closed without merging**